### PR TITLE
Allow use with the Shadow DOM disabled

### LINKED ISSUE #33: Uncaught TypeError: Cannot read property 'querySelector' of null
Uncaught TypeError: Cannot read property 'querySelector' of null

**Atom Version**: 0.158.0
**System**: Mac OS X 10.11.1
**Thrown From**: [activate-power-mode](https://github.com/JoelBesada/activate-power-mode) package, v0.3.2

### Steps To Reproduce

1. ...
2. ...

### Stack Trace

```
At /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:121

TypeError: Cannot read property 'querySelector' of null
  at Object.module.exports.ActivatePowerMode.calculateCursorOffset (/Users/Sarkizz/.atom/packages/activate-power-mode/lib/activate-power-mode.coffee:35:47)
  at Object.module.exports.ActivatePowerMode.spawnParticles (/Users/Sarkizz/.atom/packages/activate-power-mode/lib/activate-power-mode.coffee:65:21)
  at Object.debounced [as throttledSpawnParticles] (/Users/Sarkizz/.atom/packages/activate-power-mode/node_modules/lodash.throttle/node_modules/lodash.debounce/index.js:182:23)
  at Object.module.exports.ActivatePowerMode.onChange (/Users/Sarkizz/.atom/packages/activate-power-mode/lib/activate-power-mode.coffee:48:6)
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at TextBuffer.module.exports.TextBuffer.applyPatch (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:590:20)
  at TextBuffer.module.exports.TextBuffer.setTextInRange (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:503:12)
  at Selection.module.exports.Selection.insertText (/Applications/Atom.app/Contents/Resources/app/src/selection.js:446:43)
  at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:738:31
  at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:777:27
  at History.module.exports.History.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:112:18)
  at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:683:27)
  at TextEditor.module.exports.TextEditor.transact (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1146:26)
  at TextEditor.module.exports.TextEditor.mutateSelectedText (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:770:19)
  at TextEditor.module.exports.TextEditor.insertText (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:735:21)
  at TextEditor.object.(anonymous function) [as insertText] (/Applications/Atom.app/Contents/Resources/app/node_modules/underscore-plus/lib/underscore-plus.js:73:27)
  at /Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:653:23
  at History.module.exports.History.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:112:18)
  at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:683:27)
  at TextEditor.module.exports.TextEditor.transact (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1146:26)
  at [object Object].module.exports.React.createClass.onTextInput (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:652:30)
  at HTMLDivElement.boundMethod (/Applications/Atom.app/Contents/Resources/app/node_modules/react-atom-fork/lib/ReactCompositeComponent.js:1295:21)

```

/cc @atom/core

### PR DESCRIPTION
**Pull request by crazymykl**
Fixes #33

### CODE REVIEW

**Comment by m-bodmer** [comment]
+1... Solves the issue I was having which is documented in #33

**Comment by lexcast** [comment]
@crazymykl can you update this one? would be a great addition

**Comment by crazymykl** [comment]
Rebased and ready to go.

File: lib/power-canvas.coffee [hunk 1]
```diff
@@ -49,7 +49,7 @@ module.exports =
 
   getColorAtPosition: (left, top) ->
     offset = @editorElement.getBoundingClientRect()
-    el = atom.views.getView(@editor).shadowRoot.elementFromPoint(
+    el = (atom.views.getView(@editor).shadowRoot ? document).elementFromPoint(
```
Comment (lexcast) [inline lib/power-canvas.coffee]: why document here? 
mm I don't know why are we getting the editor view `@editorElement` again if it is above :confused:

Comment (crazymykl) [inline lib/power-canvas.coffee]: `elementFromPoint` is only defined on documents, per the spec (a shadow root is like a pseudodocument).

**Comment by lexcast** [comment]
thanks :tada:

**Pull request merged**